### ISSUE TITLE
Object Detection Training process was killed without throwing any problems (not RAM overflow problem)

### ISSUE THREAD

**Issue by dtlam26** [tensorflow/models#10326]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [ x] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [ x] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [ x] I checked to make sure that this issue has not already been filed.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/research/object_detection

## 2. Describe the bug

I follow the [tutorial Object Detection API](https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/install.html) and retrain data with COCO datasets. The training starts normally and logout output loss and suddenly got killed. 
```
Instructions for updating:
Use fn_output_signature instead
INFO:tensorflow:batch_all_reduce: 269 all-reduces with algorithm = nccl, num_packs = 1
I1021 19:29:20.140357 140211062587968 cross_device_ops.py:900] batch_all_reduce: 269 all-reduces with algorithm = nccl, num_packs = 1
INFO:tensorflow:batch_all_reduce: 269 all-reduces with algorithm = nccl, num_packs = 1
I1021 19:29:38.875784 140211062587968 cross_device_ops.py:900] batch_all_reduce: 269 all-reduces with algorithm = nccl, num_packs = 1
INFO:tensorflow:batch_all_reduce: 269 all-reduces with algorithm = nccl, num_packs = 1
I1021 19:29:56.933533 140211062587968 cross_device_ops.py:900] batch_all_reduce: 269 all-reduces with algorithm = nccl, num_packs = 1
INFO:tensorflow:batch_all_reduce: 269 all-reduces with algorithm = nccl, num_packs = 1
I1021 19:30:16.020746 140211062587968 cross_device_ops.py:900] batch_all_reduce: 269 all-reduces with algorithm = nccl, num_packs = 1
2021-10-21 19:30:31.649010: W tensorflow/core/grappler/optimizers/loop_optimizer.cc:906] Skipping loop optimization for Merge node with control input: replica_1/train_input_images/write_summary/summary_cond/branch_executed/_1677
INFO:tensorflow:Step 100 per-step time 2.387s
I1021 19:33:04.488567 140211062587968 model_lib_v2.py:698] Step 100 per-step time 2.387s
INFO:tensorflow:{'Loss/classification_loss': 0.6879406,
 'Loss/localization_loss': 0.29040653,
 'Loss/regularization_loss': 0.27075908,
 'Loss/total_loss': 1.2491063,
 'learning_rate': 0.014666351}
I1021 19:33:04.489341 140211062587968 model_lib_v2.py:701] {'Loss/classification_loss': 0.6879406,
 'Loss/localization_loss': 0.29040653,
 'Loss/regularization_loss': 0.27075908,
 'Loss/total_loss': 1.2491063,
 'learning_rate': 0.014666351}
INFO:tensorflow:Step 200 per-step time 1.211s
I1021 19:35:05.613877 140211062587968 model_lib_v2.py:698] Step 200 per-step time 1.211s
INFO:tensorflow:{'Loss/classification_loss': 0.5450624,
 'Loss/localization_loss': 0.262483,
 'Loss/regularization_loss': 0.2717826,
 'Loss/total_loss': 1.0793281,
 'learning_rate': 0.0159997}
I1021 19:35:05.614578 140211062587968 model_lib_v2.py:701] {'Loss/classification_loss': 0.5450624,
 'Loss/localization_loss': 0.262483,
 'Loss/regularization_loss': 0.2717826,
 'Loss/total_loss': 1.0793281,
 'learning_rate': 0.0159997}
INFO:tensorflow:Step 300 per-step time 1.216s
I1021 19:37:07.180053 140211062587968 model_lib_v2.py:698] Step 300 per-step time 1.216s
INFO:tensorflow:{'Loss/classification_loss': 0.5462525,
 'Loss/localization_loss': 0.30112177,
 'Loss/regularization_loss': 0.27352968,
 'Loss/total_loss': 1.120904,
 'learning_rate': 0.01733305}
I1021 19:37:07.180727 140211062587968 model_lib_v2.py:701] {'Loss/classification_loss': 0.5462525,
 'Loss/localization_loss': 0.30112177,
 'Loss/regularization_loss': 0.27352968,
 'Loss/total_loss': 1.120904,
 'learning_rate': 0.01733305}
Killed

```
I have followed s[this issue](`https://github.com/tensorflow/models/issues/3497`) and check `dmesg -k`
```

[3138705.607699] test1[33417]: segfault at 656c6c6f ip 000055d26855077c sp 00007ffff2eeaab0 error 4 in test1[55d268550000+1000]
[3138765.887948] test1[33660]: segfault at 656c6c6f ip 000055f1846a677c sp 00007ffcde90ad50 error 4 in test1[55f1846a6000+1000]
[3138772.551306] test1[33687]: segfault at 656c6c6f ip 000055832d83077c sp 00007ffc23668210 error 4 in test1[55832d830000+1000]
[3138783.897230] NVRM: Xid (PCI:0000:18:00): 31, pid=33750, Ch 00000039, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_3 faulted @ 0x7f75_7bd4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3138883.734400] NVRM: Xid (PCI:0000:18:00): 31, pid=34180, Ch 00000039, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_8 faulted @ 0x7f69_43d4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3138904.688266] NVRM: Xid (PCI:0000:18:00): 31, pid=34292, Ch 00000039, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_6 faulted @ 0x7f1b_fdd4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3138940.219240] NVRM: Xid (PCI:0000:18:00): 31, pid=34464, Ch 00000039, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_0 faulted @ 0x7f8a_cbd4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3139006.882455] NVRM: Xid (PCI:0000:18:00): 31, pid=34758, Ch 00000039, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_7 faulted @ 0x7fab_7dd4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3139018.472569] NVRM: Xid (PCI:0000:18:00): 31, pid=34833, Ch 00000039, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_7 faulted @ 0x7f37_ebd4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3139872.788935] NVRM: Xid (PCI:0000:18:00): 31, pid=36654, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_2 faulted @ 0x7fc6_17d4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3140104.167805] NVRM: Xid (PCI:0000:18:00): 31, pid=36824, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_6 faulted @ 0x7fc1_efd4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3140143.986837] NVRM: Xid (PCI:0000:18:00): 31, pid=36855, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_10 faulted @ 0x7ff7_55d4e000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3140588.899090] python[37252]: segfault at 0 ip 00007feab0e91650 sp 00007fff1aeb0a00 error 4 in _pywrap_tensorflow_internal.so[7feaae2c8000+2d5be000]
[3200414.809015] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3200414.809021] NVRM: Xid (PCI:0000:18:00): 31, pid=36174, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_7 faulted @ 0x7f95_87d4e000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3200459.016817] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3200459.016823] NVRM: Xid (PCI:0000:18:00): 31, pid=36233, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_3 faulted @ 0x7f76_add4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3200526.917934] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3200526.917940] NVRM: Xid (PCI:0000:18:00): 31, pid=36296, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_7 faulted @ 0x7f14_afd4e000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3200545.094503] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3200545.094509] NVRM: Xid (PCI:0000:18:00): 31, pid=36358, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_8 faulted @ 0x7f16_13d4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3204617.861950] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3204617.861955] NVRM: Xid (PCI:0000:18:00): 31, pid=36487, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_4 faulted @ 0x7f59_3bd4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3204644.132369] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3204644.132371] NVRM: Xid (PCI:0000:18:00): 31, pid=36546, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_0 faulted @ 0x7f90_b9d4f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3204837.137552] pynet-0.1.out[36619]: segfault at 18 ip 0000558537eb4b32 sp 00007fffb7d59700 error 4 in pynet-0.1.out[558537e94000+b1000]
[3205513.444556] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3205513.444562] NVRM: Xid (PCI:0000:18:00): 31, pid=37214, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_6 faulted @ 0x7fc3_8120f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3205973.213766] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3205973.213771] NVRM: Xid (PCI:0000:18:00): 31, pid=40680, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_5 faulted @ 0x7f85_9d20f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3205994.552125] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3205994.552131] NVRM: Xid (PCI:0000:18:00): 31, pid=40803, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_6 faulted @ 0x7fdf_f5214000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3206021.679152] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3206021.679157] NVRM: Xid (PCI:0000:18:00): 31, pid=40953, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_3 faulted @ 0x7f94_7320f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3206145.037404] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3206145.037410] NVRM: Xid (PCI:0000:18:00): 31, pid=41520, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_6 faulted @ 0x7f50_4320e000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3206159.414010] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3206159.414016] NVRM: Xid (PCI:0000:18:00): 31, pid=41578, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_7 faulted @ 0x7f4b_fd20f000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3206177.371201] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3206177.371206] NVRM: Xid (PCI:0000:18:00): 31, pid=41637, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_8 faulted @ 0x7f01_c320e000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3206309.157566] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3206309.157572] NVRM: Xid (PCI:0000:18:00): 31, pid=42218, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_4 faulted @ 0x7ff5_52c00000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3206327.188774] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3206327.188781] NVRM: Xid (PCI:0000:18:00): 31, pid=42289, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_10 faulted @ 0x7f5a_62c00000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3206470.823663] NVRM: GPU at PCI:0000:18:00: GPU-d7d12c0c-9406-6ae6-beea-8ed0d8d58a71
[3206470.823669] NVRM: Xid (PCI:0000:18:00): 31, pid=42415, Ch 00000008, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_10 faulted @ 0x7fb6_1b20e000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE
[3213651.471549] NVRM: GPU at PCI:0000:3b:00: GPU-e67549a6-5464-541e-2f3b-01ffc864170e
[3213651.471555] NVRM: Xid (PCI:0000:3b:00): 31, pid=61855, Ch 0000000b, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_0 faulted @ 0x0_00004000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_WRITE

```
## 3. Steps to reproduce

I am not root user, so have to use conda to configurate everything.

download cuda toolkit 11.2 from conda-forge and install
`cuda install path_to_package`
download cudnn from nvidia and export path
install tensorflow 2.5
run training from Object Detection API

## 4. Expected behavior

Training got killed, but htop  show no overflow in RAM. 

## 5. Additional context

Tensorflow operate normally, can inference model from checkpoint. GPU is available.
I have tried to use different models and tensorflow version, but nothing work out. I have tried with different python (3.8)

## 6. System information

- OS Platform and Distribution: Linux Ubuntu Bionic 18.04
- Mobile device name if the issue happens on a mobile device:
- TensorFlow installed from pip
- TensorFlow version (use command below): 2.5
- Python version: 3.9
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source): 2.7
- CUDA/cuDNN version: 11.2 - 8.2.1
- GPU model and memory:

**Comment by dtlam26** [comment]
The problem is because of the slum job
